balances-adjustment/src/test/java/com/powsybl/balances_adjustment/balance_computation/extension/BalanceComputationParametersExtension.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) 2023, Coreso SA (https://www.coreso.eu/) and TSCNET Services GmbH (https://www.tscnet.eu/)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.balances_adjustment.balance_computation.extension;
+
+import com.powsybl.balances_adjustment.balance_computation.BalanceComputationParameters;
+import com.powsybl.commons.extensions.AbstractExtension;
+import com.powsybl.loadflow.LoadFlowResult;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Marine Guibert {@literal <marine.guibert at artelys.com>}
+ * @author Damien Jeandemange {@literal <damien.jeandemange at artelys.com>}
+ */
+public class BalanceComputationParametersExtension extends AbstractExtension<BalanceComputationParameters> {
+
+    private final List<LoadFlowResult> loadFlowResultsPerIteration = new ArrayList<>();
```

Comment:
I really don't get what you are doing here. This is an extension of balance computation parameters but you seem to put inside:
- A list of `LoadFlowResult`: a natural design is to increase the `BalanceComputationResult`, no? And it will benefit everybody. If you think that it is not generic enough because we can run a balance ajustement with only one call to a loadflow, we can think about a more generic design all together.
- A list of mismatches: what is the definition of a mismatch for you? It is also a result, so why putting that inside parameters.
Even if it is just for test, affordance should be a guide line. 

Furthermore, it seems (to be confirmed by @phiedw) that you add an implementation of  `BalanceComputation` but for me the issue is maybe more linked to the lack of genericity of the current one.

Reply:
I agree with this comment. It is very odd to create an extension of parameters to store intermediate results inside of it. It would indeed make more sense to add information to the BalanceComputationResult instead. If this is not currently possible with the implementations we have then we should maybe change the design of some of these classes.

Reply:
`BalanceComputationParametersExtension` could have been better renamed `BalanceComputationResultsRecorder`.

Currently there is no way to extend BalanceComputationResult and this would require some other changes.  
Workaround is here to use a 'recorder' transmitted via parameters extension.

If you don't like it I can just remove this example, and we may work our a design in another PR

Reply:
What is the issue with putting these fields inside `BalanceComputationResult`?

Reply:
Issue is that I think it is up to the library user to choose what they want to do, and that we should not enforce anything.

The results stored here are only a test/demo. Library users may or may not want to do this.  

They may want to do other things like writing iIDM at each iteration, creating/copying current variant to another one, retrying load flow with other load flow options, etc....